Tests/Unit/Repository/RepositoryTestCase.php

```diff
@@ -11,10 +11,36 @@
 
 namespace Symfony\Cmf\Component\Resource\Tests\Unit\Repository;
 
-use Prophecy\PhpUnit\ProphecyTestCase;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
 
-abstract class RepositoryTestCase extends ProphecyTestCase
+abstract class RepositoryTestCase extends \PHPUnit_Framework_TestCase
 {
+    protected $documentManager;
+    protected $managerRegistry;
+    protected $childrenCollection;
+    protected $finder;
+    protected $uow;
+    protected $document;
+    protected $child1;
+    protected $child2;
+    protected $object;
+    protected $resource;
+    protected $session;
+    protected $node;
+    protected $rootNode;
+
+    public function setUp()
+    {
+        $this->session = $this->prophesize('PHPCR\SessionInterface');
+        $this->finder = $this->prophesize('DTL\Glob\FinderInterface');
+        $this->node = $this->prophesize('PHPCR\NodeInterface');
+        $this->node2 = $this->getMock('PHPCR\NodeInterface');
```

Comment:
You probably need to call ->reveal() to get the actual ''mock" and not the prophecy.

On 21 May 2016 12:45:19 BST, Maximilian Berghoff <notifications@github.com> wrote:
>> +    protected $uow;
>> +    protected $document;
>> +    protected $child1;
>> +    protected $child2;
>> +    protected $object;
>> +    protected $resource;
>> +    protected $session;
>> +    protected $node;
>> +    protected $rootNode;
>> +
>> +    public function setUp()
>> +    {
>> +        $this->session =
>$this->prophesize('PHPCR\SessionInterface');
>> +        $this->finder =
>$this->prophesize('DTL\Glob\FinderInterface');
>> +        $this->node = $this->prophesize('PHPCR\NodeInterface');
>> +        $this->node2 = $this->getMock('PHPCR\NodeInterface');
>
>The expectected interface does not match when using ProphetObject. Got
>an error in the PhpcResource constructor and no match on instanceOf. I
>think it's is a commun usage in phpspec Not to mock interfaces using
>Test implementations instead. I did that for the document. But my node
>implementation can't be mocked as i got strange errors Last Night. 
>
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/symfony-cmf/resource/pull/17/files/9448e050eef5d1120671057c950a293409451260#r64133327

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.